pypeit/flatfield.py

```diff
@@ -109,7 +117,7 @@ def from_master_file(cls, master_file, par=None):
 
     def __init__(self, spectrograph, par, files=None, det=1, master_key=None,
```

Comment:
While we are reworking this, let's think through the calling sequence. I think this code has no use without most of the variables which are optional. If I'm wrong there, show me the use cases and the calling sequence. Otherwise, let's make the required quantities to make the flat required variables. 

It would be highly preferable to be able to run this code on a file assuming that the other calibrations exist, i.e. read slits and tilts from masters then execute this module by passing in the relevant quantities and the filename.

Reply:
It's useful to bring this up, but I think it goes beyond the scope of this PR.  You should raise this as an issue, and let's talk about it at an upcoming meeting.

Reply:
I am happy to chat about this, but am sticking to my guns here. Anything that this code does that does not involve making a flat should be a classmethod, i.e. load save. The optional parameters are not optional, so I think tilts and slits need to be required parameters like spectrograph.

Reply:
@jhennawi , I think we decided to revisit this in a different PR, right?

Reply:
Yes that is fine @kbwestfall, but beware @profxj that these infinitely long optional argument lists will be a thing of the past soon.